modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/AddObjectFieldCompositeKeyCandidatesMVCResourceCommand.java

```diff
@@ -0,0 +1,171 @@
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Leite
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ObjectPortletKeys.OBJECT_DEFINITIONS,
+		"mvc.command.name=/object_definitions/add_object_field_composite_key_candidates"
+	},
+	service = MVCResourceCommand.class
+)
+public class AddObjectFieldCompositeKeyCandidatesMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		long objectDefinitionId = ParamUtil.getLong(
+			resourceRequest, "objectDefinitionId");
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				objectDefinitionId);
+
+		User user = _userLocalService.getUser(objectDefinition.getUserId());
+
+		List<String> objectFieldLabels = new ArrayList<>();
+
+		for (Long objectFieldId :
+				ParamUtil.getLongValues(resourceRequest, "objectFieldsIds")) {
+
+			ObjectField objectField = _objectFieldLocalService.fetchObjectField(
+				objectFieldId);
+
+			try {
+				Table<?> table = _objectFieldLocalService.getTable(
+					objectDefinitionId, objectField.getName());
+
+				Column<?, ?> column = table.getColumn(
+					objectField.getDBColumnName());
+
+				long count = _objectEntryLocalService.getObjectEntriesCount(
+					0, objectDefinition, column.isNotNull());
+
+				if (count == 0) {
+					continue;
+				}
+
+				objectFieldLabels.add(objectField.getLabel(user.getLocale()));
+			}
+			catch (PortalException portalException) {
+				SessionErrors.add(resourceRequest, portalException.getClass());
+			}
+		}
+
+		JSONPortletResponseUtil.writeJSON(
+			resourceRequest, resourceResponse,
+			JSONUtil.put(
+				"errorLabel",
+				() -> {
+					if (objectFieldLabels.isEmpty()) {
+						return StringPool.BLANK;
+					}
+
+					String fieldKey = "field";
+					String objectFieldLabelsArgument = objectFieldLabels.get(0);
+
+					if (objectFieldLabels.size() > 1) {
+						fieldKey = "fields";
+
+						StringBuilder stringBuilder = new StringBuilder(
+							objectFieldLabels.get(0));
+
+						for (int i = 1; i < (objectFieldLabels.size() - 1);
+							 i++) {
+
+							stringBuilder.append(StringPool.COMMA_AND_SPACE);
+							stringBuilder.append(objectFieldLabels.get(i));
+						}
+
+						stringBuilder.append(StringPool.SPACE);
+						stringBuilder.append(
+							_language.get(
+								_portal.getHttpServletRequest(resourceRequest),
+								"and"
+							).toLowerCase());
```

Comment:
@pedro-oliveira446 this is also very strange to me. Do we really need this `and` here? And how can we be sure that this will work for all languages ​​on the portal (like chinese or japanese, for example)?